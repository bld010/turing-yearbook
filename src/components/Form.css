.Form {
  height: auto;
  width: 300px;
  display: flex;
  flex-direction: column;
  margin: auto;

}