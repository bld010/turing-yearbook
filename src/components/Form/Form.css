.Form {
  height: auto;
  width: 200px;
  display: flex;
  flex-direction: column;
  margin: auto;
  padding: 0px 20px 20px 0;

}