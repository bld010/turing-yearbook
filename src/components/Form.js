import React, { Component } from 'react';
import './Form.css'

class Form extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      quote: '',
      superlative: '',
      photo: 'https://placekitten.com/200/300'
    }
  }

  handleChange = e => {
    this.setState({[e.target.name]: e.target.value})
  }

  handleAddNewStudent = (e) => {
    e.preventDefault();
    this.props.addNewStudent(this.state);
    this.setState({
      name: '',
      quote: '',
      superlative: ''
    })
  }

  render() {
    return(
      <form class="Form">
        <input
          type="text"
          name="name"
          placeholder="Name"
          value={this.state.name}
          onChange={this.handleChange}
        />
        <input
          type="text"
          name="quote"
          placeholder="Quote"
          value={this.state.quote}
          onChange={this.handleChange}
        />
        <input
          type="text"
          name="superlative"
          placeholder="Superlative"
          value={this.state.superlative}
          onChange={this.handleChange}
        />
        <button onClick={this.handleAddNewStudent}>Add Student</button>
      </form>
    ) 
  }
}


export default Form;