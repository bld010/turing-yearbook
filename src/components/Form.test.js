import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

it('should match the snapshot', () => {
  const wrapper = shallow(<Form />)

  expect(wrapper).toMatchSnapshot();
})

//Test each input contenteditable

//Test each input onChange firing handleChange

//Test button click firing handleAddNewStudent

//Test handleChange