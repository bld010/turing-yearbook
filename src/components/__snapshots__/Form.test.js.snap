// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<form
  class="Form"
>
  <input
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    name="quote"
    onChange={[Function]}
    placeholder="Quote"
    type="text"
    value=""
  />
  <input
    name="superlative"
    onChange={[Function]}
    placeholder="Superlative"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Add Student
  </button>
</form>
`;
